plugins/out_stackdriver/stackdriver_resource_types.c

```diff
@@ -0,0 +1,139 @@
+    },
+    { 
+        .id = RESOURCE_TYPE_GENERIC_TASK,
+        .resources = {"generic_task"},
+        .required_labels = {"location", "namespace", "job", "task_id"}
+    }
+};
+
+static char **get_required_labels(int resource_type)
+{
+    int i;
+    int len;
+
+    len = sizeof(resource_types) / sizeof(resource_types[0]);
+    for(i = 0; i < len; i++) {
+        if (resource_types[i].id == resource_type) {
+            return (char **) resource_types[i].required_labels;
+        }
+    }
+    return NULL;
+}
+
+/*
+ *   set_resource_type():
+ * - Iterates through resource_types that are set up for validation and sets the 
+ *   resource_type if it matches one of them.
+ */
+void set_resource_type(struct flb_stackdriver *ctx)
+{
+    int i;
+    int j;
+    int len;
+    char *resource;
+    struct resource_type resource_type;
+
+    len = sizeof(resource_types) / sizeof(resource_types[0]);
+    for(i = 0; i < len; i++) {
+        resource_type = resource_types[i];
+        for(j = 0; j < MAX_RESOURCE_ENTRIES; j++) {
+            if (resource_type.resources[j] != NULL) {
+                resource = resource_type.resources[j];
+                if (flb_sds_cmp(ctx->resource, resource, strlen(resource)) == 0) {
+                    ctx->resource_type = resource_type.id;
+                    return;
+                }
+            }
+        }
+    }
+}
+
+/*
+ *   resource_api_has_required_labels():
+ * - Determines if all required labels for the set resource type are present as
+ *   keys on the resource labels key-value pairs.
+ */
+int resource_api_has_required_labels(struct flb_stackdriver *ctx)
+{
+    struct mk_list *head;
+    struct flb_hash *ht;
+    struct flb_kv *label_kv;
+    char** required_labels;
+    int i;
+    int found;
+    void *tmp_buf;
+    size_t tmp_size;
+
+    if (mk_list_size(&ctx->resource_labels_kvs) == 0) {
+        return FLB_FALSE;
+    }
+
+    required_labels = get_required_labels(ctx->resource_type);
+    if (required_labels == NULL) {
+        return FLB_FALSE;
+    }
+
+    ht = flb_hash_create(FLB_HASH_EVICT_NONE, MAX_REQUIRED_LABEL_ENTRIES, 0);
+    mk_list_foreach(head, &ctx->resource_labels_kvs) {
+        label_kv = mk_list_entry(head, struct flb_kv, _head);
+        for (i = 0; i < MAX_REQUIRED_LABEL_ENTRIES; i++) {
+            if (required_labels[i] != NULL && flb_sds_cmp(label_kv->key,
+                required_labels[i], strlen(required_labels[i])) == 0) {
+                flb_hash_add(ht, required_labels[i], strlen(required_labels[i]), 
+                    NULL, 0);
+            }
+        }
+    }
+
+    for (i = 0; i < MAX_REQUIRED_LABEL_ENTRIES; i++) {
+        if (required_labels[i] != NULL) {
+            found = flb_hash_get(ht, required_labels[i], strlen(required_labels[i]),
+                &tmp_buf, &tmp_size);
+            if (found == -1) {
+                flb_plg_warn(ctx->ins, "the resource labels api will not be used, " 
```

Comment:
Instead of saying "the resource labels api will not be used", we should directly use the option name `resource_labels` here to make it more clear. We could say something like:

"labels set in resource_labels will not be applied"